src/services/casbin/casbin.conf

```diff
@@ -0,0 +1,14 @@
+[request_definition]
+r = sub, res, act
+
+[policy_definition]
+p = role, scope, res, act
```

Comment:
what is the intention of scope here @bodymindarts ?

this is a non standard casbin.conf so I'm trying to understand why you went that path instead of using one of the proposed "standard" RBAC conf file

Reply:
This is [one of the standard configs](https://casbin.org/docs/en/rbac-with-domains).

Why the 'scope'? I think it is appropriate for our system to consider the `account` as the `domain` / `scope` as described in the example. Consider the following use cases:

A user who should have complete access to all wallets of an account (ie account admin?).
Without the scope every time a new wallet is created in an account we would have to persist an association triplet as follows:
`subject: user/123` - `permission: wallet/read` - `resource: wallet/456` - `allow`.
When using `account` as the scope we could also do the following:
`subject: user/123` - `scope: account/456` - `permission: wallet/read` - `resource: wallet/*` - `allow`.

Or more realistically:
`role: account/admin` - `scope: account/456` - `permission: wallet/read` - `resource: wallet/*` - `allow`.

and then assign the `user/123` to `role: account/admin` in `scope: account/456`.

Then any user with the `account/admin` role would inherit access to any additional wallet created under `account/456` without needing to persist an additional triplet.

Put another way without the `scope` `wallet/*` would give permission to absolutely every wallet in the system _with_ `scope` `wallet/*` can give access to any wallet _of a specific account_ (aka scope / domain).